connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -817,6 +818,11 @@ public void onCompletion(Throwable error, Void result) {
     // by the system, then checks whether any configs have actually changed before submitting the new configs to storage
     private void reconfigureConnector(final String connName, final Callback<Void> cb) {
         try {
+            if (configState.targetState(connName) == TargetState.PAUSED) {
+                log.trace("Skipping reconfiguration of paused connector {}", connName);
```

Comment:
Only trace level so I'm not too worried about it, but this could probably be a bit clearer -- this should really only happen if previous configuration failed, we had to retry, and the state has then changed to PAUSED. We might want to explain here that this indicates there was a previous failure, we're giving up reconfiguring due to the target state change, and that when its resumed we'll try reconfiguring again.